### PR TITLE
Add validation to log_level and url

### LINKED ISSUE #23: Data source input validation
There are some fields such as `log_level`, `path` and `url` that expect data in a certain format. The Generate method should catch these errors during the apply sequence, but it would be a lot better if it was validated a head of time.

Fields that need validation should have a validate func that enforces restrictions on the input.
https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#validatefunc

### PR DESCRIPTION
**Pull request by nlarzon**
If this PR is accepted the log_level, url and arch field will be validated

Partially fixes #23

### CODE REVIEW

**Comment by phillebaba** [comment]
@nlarzon nice work <3

Could we add additional acceptance tests that check these validators?

File: pkg/provider/data_install.go [hunk 1]
```diff
@@ -92,10 +93,11 @@ func DataInstall() *schema.Resource {
...
 			"log_level": {
-				Description: "Log level for toolkit components.",
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     installDefaults.LogLevel,
+				Description:  "Log level for toolkit components.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      installDefaults.LogLevel,
+				ValidateFunc: validation.StringInSlice([]string{"info", "debug", "error"}, false),
```
Comment (phillebaba) [inline pkg/provider/data_install.go:100]: Is it possible to get these values from a package in flux2?

Comment (stefanprodan) [reply pkg/provider/data_install.go:100]: No, but we could expose the args accepted values in a package. I would merge the validation and refactor later.

Comment (nlarzon) [reply pkg/provider/data_install.go:100]: I tried that but they are not exposed as a pkg so I couldn't get at them, maybe a more experienced gopher could.

**Comment by stefanprodan** [comment]
@nlarzon can you please add a validation for arch it's `amd64, arm64, arm`.

**Comment by nlarzon** [comment]
I will try to add the test.

File: pkg/provider/data_install.go (hunk 1 above)
Comment (phillebaba) [reply pkg/provider/data_install.go:101]: I will create an issue to track this so we can merge this first

**Comment by phillebaba** [comment]
@nlarzon could you rebase with the latest changes from main? After that I am fine to merge this.

**Review by phillebaba — APPROVED** [review]
LGTM :medal_sports: 

I ran through the example without any issue. We can merge this and I will add e2e tests for the future to verify the provider.

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**